python/adelphi/tests/integration/test_cql.py

```diff
@@ -64,19 +73,43 @@ def buildSchema(self):
 
 
     def compareToReferenceCql(self, referencePath, comparePath):
-        referenceSet = digestSet(referencePath)
-        compareSet = digestSet(comparePath)
+        compareLines = linesWithNewline(comparePath)
+        referenceLines = linesWithNewline(referencePath)
+
+        diffGen = difflib.unified_diff(
+            compareLines,
+            referenceLines,
+            fromfile=os.path.abspath(comparePath),
+            tofile=os.path.abspath(referencePath))
+
+        diffEmpty = True
+        for line in diffGen:
+            if diffEmpty:
+                print("Diff of generated file ({}) against reference file ({})".format(
+                    os.path.basename(comparePath),
+                    os.path.basename(referencePath)))
+            diffEmpty = False
+            print(line.strip())
 
-        refOnlySet = referenceSet - compareSet
-        if len(refOnlySet) > 0:
-            log.info("Statements in reference file {} but not in compare file {}:".format(referencePath, comparePath))
-            logCqlDigest(referencePath, refOnlySet)
-        compareOnlySet = compareSet - referenceSet
-        if len(compareOnlySet) > 0:
-            log.info("Statements in compare file {} but not in reference file {}:".format(comparePath, referencePath))
-            logCqlDigest(comparePath, compareOnlySet)
+        if not diffEmpty:
+            self.fail()
 
-        self.assertEqual(referenceSet, compareSet)
+
+    def combineSchemas(self):
+        outputDirPath = self.outputDirPath(self.version)
+        allOutputFileName = "{}-all".format(self.version)
+        allOutputPath = self.outputDirPath(allOutputFileName)
+
+        schemaPaths = glob.glob("{}/*/schema".format(outputDirPath))
+        self.assertGreater(len(schemaPaths), 0)
+        schemas = { extractKeyspaceName(p) : p for p in schemaPaths}
+        sortedKeyspaces = sorted(schemas.keys())
+
+        with open(allOutputPath, "w+") as allOutputFile:
+            cqlStr = "\n\n".join(open(schemas[ks]).read() for ks in sortedKeyspaces)
+            allOutputFile.write(cqlStr)
+
+        return allOutputPath
```

Comment:
Another problem observed in testing: the ordering of individual keyspace schemas returned by the glob above isn't deterministic so there's no guarantee we'll get the right ordering of keyspaces.  To fix this we extract the keyspace name from each individual keyspace schema and traverse them in order while generating the "all" schema.